ui/ducks/send/send.js

```diff
@@ -2128,6 +2142,27 @@ export function signTransaction() {
         ),
       };
 
+      // If, after edit, we are sending the native asset,
+      //   and if the send duck hex data was reset on this edit,
+      //   and if the hex data from the tx's background state matches
+      //   one of the special token methods that we handle,
+      //   then we conclude that the user does not want to send hex data,
+      //   and ensure it gets reset in the background state as well.
+      const sendingNativeAsset =
+        draftTransaction.asset.type === ASSET_TYPES.NATIVE;
+      const userInputHexDataHasBeenReset =
+        draftTransaction.userInputHexData === '';
+      const priorHexDataMatchesSpecialTokenTxSignatures =
+        parseStandardTokenTransactionData(editingTx.txParams.data) !==
+        undefined;
+      if (
+        sendingNativeAsset &&
+        userInputHexDataHasBeenReset &&
+        priorHexDataMatchesSpecialTokenTxSignatures
+      ) {
+        editingTx.txParams.data = '';
+      }
```

Comment:
I"m confused why we can't just rely upon the value of userInputHexData to override this? It should be set to an empty string n 1920, therefor the call to generateTransactionParams on line 2115 should return an appropriate txParams object. with data null. Can you help me understand why this contract breaks down under these special conditions?

Reply:
yes removed in ee260f475